src/frontend/src/components/AddressAutocomplete.tsx

```diff
@@ -0,0 +1,78 @@
+import { throttle } from "lodash";
+import { useCallback, useRef } from "react";
+import { useParams } from "react-router";
+
+import { getAddressSuggestions } from "api/addressSearch";
+import {
+    Autocomplete,
+    type Suggestion,
+} from "components/base/Autocomplete/Autocomplete";
+import { Language, type LanguageKey } from "src/enums";
+
+const RATE_LIMIT = 500;
+
+const AddressAutocomplete = ({
+    placeholder,
+    onClearCallback,
+    onSuggestionCallback,
+}: {
+    placeholder: string;
+    onClearCallback: () => void;
+    onSuggestionCallback: (suggestion: Suggestion) => void;
+}) => {
+    const { lang } = useParams<{ lang: LanguageKey | undefined }>();
+    const abortRef = useRef<AbortController | null>(null);
+    const cacheRef = useRef<Record<string, Suggestion[]>>({});
+
+    // When autocomplete is enabled in forward geocoding query,
+    // each user keystroke counts as one request to the Geocoding API.
+    // Implement strategies from old site to reduce requests.
+    // https://github.com/azavea/echo-locator/blob/master/taui/src/components/geocoder.js#L81-L105
+    const fetchSuggestions = useCallback(
+        async (query: string, signal: AbortSignal) => {
+            if (!query.trim()) return [];
+            // Return cached result if available
+            if (cacheRef.current[query]) {
+                return cacheRef.current[query];
+            }
+
+            // Abort previous request
+            if (abortRef.current) {
+                abortRef.current.abort();
+            }
+
+            const controller = new AbortController();
+            abortRef.current = controller;
+
+            return await getAddressSuggestions(
```

Comment:
I understand this block was likely adapted from the old code base, but did you get a chance to try the following pattern which follows closer to what we do now in this new version of the app?
```ts
const { data: { features } } = await getAddressSuggestions(...);
return features.map(...);
```